### PR TITLE
meow meow meow

### PR DESCRIPTION
**Pull request by kaatrasalo**
looksinside

### CODE REVIEW

File: handling.go [hunk 1]
```diff
@@ -154,7 +156,16 @@ func processRequest(c *gin.Context, collageImages bool, downloadSound bool) (Sim
 	if skip {
 		return tiktokData, true, nil
 	}
-	tiktokData.DownloadImages()
+	var failedImageCount uint8 = tiktokData.DownloadImages()
```
Comment (Masterjoona) [inline handling.go]:
```suggestion
	failedImageCount := tiktokData.DownloadImages()
```

File: http.go [hunk 2]
```diff
@@ -76,10 +77,12 @@ func downloadImage(url string) ([]byte, error) {
 	return imageBytes, nil
 }
 
-func (t *SimplifiedData) DownloadImages() {
+func (t *SimplifiedData) DownloadImages() uint8 {
 	var wg sync.WaitGroup
 	var indexedImages []ImageWithIndex
 
+	var failedCount uint8 = 0 // Max images in a tiktok is 35, so we can expect fails to be <=35
```
Comment (Masterjoona) [inline http.go]:
```suggestion
	var failedCount uint8 // Max images in a tiktok is 35, so we can expect fails to be <=35
```

File: handling.go [hunk 3]
```diff
@@ -154,7 +156,16 @@ func processRequest(c *gin.Context, collageImages bool, downloadSound bool) (Sim
...
+	if failedImageCount > 0 {
+		tiktokData.Caption += fmt.Sprintf("\n\nFailed to download %d images", failedImageCount)
+		if failedImageCount == uint8(len(tiktokData.ImageLinks)) { // What could go wrong with casting to uint8 :clueless:
```
Comment (Masterjoona) [inline handling.go]: I don't like this nested if statement

**Comment by Masterjoona** [comment]
Im not too sure about the unint8. Its a micro optimization

**Comment by kaatrasalo** [comment]
I suppose we could just make it an int, at least then tiktoks that bypass limits can fail all they want

**Comment by Masterjoona** [comment]
Good, and do you want to fix the regex funnies or?

**Comment by kaatrasalo** [comment]
I'll go swap out the uint for an int, and get that nesting outta here. Could you handle your regex magic in the meanwhile?

**Comment by Masterjoona** [comment]
thanks!!

**Pull request merged**